ecl/hql/hqlfold.cpp

```diff
@@ -5183,8 +5183,12 @@ IHqlExpression * CExprFolderTransformer::doFoldTransformed(IHqlExpression * unfo
     case no_unlikely:
         {
             IHqlExpression * child = expr->queryChild(0);
-            if (child->queryValue())
-                return LINK(child);
+
+            if (child->getOperator()==no_likely || child->getOperator()==no_unlikely)
+                return createValue(op, expr->getType(), LINK(child->queryChild(0)));
```

Comment:
@shamser Hasn't this lost the likelihood associated with expr?  I would have expected

`replaceChild(expr, 0, child->queryChild(0));`